### PR TITLE
Non-Fungible Token with usage rights

### PR DESCRIPTION
**Pull request by KKimos**
This standard adds a new right of Non-Fungible Token that the right to use. Through this standard, you can achieve :

Separation of the right to use and ownership of Non-Fungible Token
Non-secured lease Non-Fungible Token
You can continue to use it after you mortgage the Non-Fungible Token
Metaverse sharing economy
It is precisely because of the separation of ownership and use right that the utilization rate of assets can be greater. You must distinguish between the rights of the user and the owner.

Different from 2615, we only added a right to use, so that we can get all the functions of 2615, and reduce many unnecessary operations in 2615. The explanation is as follows:

The owner should not use some of the ownership rights when mortgage NFT, such as transfer, AXS-like breeding or weapon upgrade, because it is likely to change the status of the original NFT, so when the user mortgages the NFT, the ownership must be mortgaged in the contract middle.
In the case of mortgage or transaction, the owner has the right to continue to use it before returning the ransom or selling it
This is the result of my implementation, you can refer to https://github.com/KKimos/ERCX

### CODE REVIEW

**Comment by alita-moore** [comment]
> A critical exception has occurred (cc @alita-moore):
> 	Message: can not read a block mapping entry; a multiline key may not be an implicit key at line 3, column 7:
>     author: kkimos <bol@zju.edu.cn>
>           ^
> 

Will address later

**Comment by KKimos** [comment]
> I have been trying but I don't know what went wrong

File: EIPS/eip-draft_Non-Fungible Token with usage rights_abbrev.md
Comment (MicahZoltu) [inline EIPS/eip-draft_Non-Fungible Token with usage rights_abbrev.md:2]:
```suggestion
title: Non-Fungible Token with usage rights
description: Separation of Non-Fungible Token ownership and usage rights , Truly in line with real life.
```
Description is required.  Simple Summary is no longer a required field.  See https://raw.githubusercontent.com/ethereum/EIPs/master/eip-template.md for an example to follow.

File: EIPS/eip-draft_Non-Fungible Token with usage rights_abbrev.md
Comment (MicahZoltu) [inline EIPS/eip-draft_Non-Fungible Token with usage rights_abbrev.md:1]:
```suggestion
---
eip: 4512
```
Please also rename the file to `EIPS/eip-4512.md`

File: EIPS/eip-draft_Non-Fungible Token with usage rights_abbrev.md
Comment (MicahZoltu) [inline EIPS/eip-draft_Non-Fungible Token with usage rights_abbrev.md:5]: There must be at least one author with a GitHub account.
```suggestion
author: KKimos (@KKimos)
```
```suggestion
author: KKimos (@KKimos), KKimos <bol@zju.edu.cn>
```
Either of the above is acceptable.

File: EIPS/eip-draft_Non-Fungible Token with usage rights_abbrev.md
Comment (MicahZoltu) [inline EIPS/eip-draft_Non-Fungible Token with usage rights_abbrev.md:14]:
```suggestion
```
This section is deprecated in favor of a `description` header field.

File: EIPS/eip-draft_Non-Fungible Token with usage rights_abbrev.md
Comment (MicahZoltu) [inline EIPS/eip-draft_Non-Fungible Token with usage rights_abbrev.md:159]:
```suggestion
```
External links should not be used.  Either inline the reference implementation, add it to `../assets/eip-4512/` or just remove this section entirely (it is optional).

Comment (KKimos) [reply EIPS/eip-draft_Non-Fungible Token with usage rights_abbrev.md]: Thanks for your reply. I have made changes but need to review the new ones. Do I need to submit them at once? Still waiting for review?

**Comment by MicahZoltu** [comment]
It looks like CI is passing, so you just have to wait for a review from lightclient or axic.  They *usually* get to it within a week, but sometimes it is faster or slower.

File: EIPS/eip-4512.md
Comment (chee-chyuan) [inline EIPS/eip-4512.md:13]: you already have description in the header. you can remove simple summary

File: EIPS/eip-4512.md
Comment (chee-chyuan) [inline EIPS/eip-4512.md:31]: do use the format `EIP-N`

File: EIPS/eip-4512.md
Comment (chee-chyuan) [inline EIPS/eip-4512.md:92]: ERC721 -> ERC-721

File: EIPS/eip-4512.md
Comment (chee-chyuan) [inline EIPS/eip-4512.md:94]: ERC721 -> ERC-721

File: EIPS/eip-4512.md
Comment (chee-chyuan) [inline EIPS/eip-4512.md:70]: personally i do feel that including doc strings such as natspec may be helping in getting the reader up to speed

File: EIPS/eip-4512.md
Comment (chee-chyuan) [inline EIPS/eip-4512.md:83]: Enumerable?

File: EIPS/eip-4512.md
Comment (chee-chyuan) [inline EIPS/eip-4512.md:29]: Airbnb

**Review by chee-chyuan — CHANGES_REQUESTED** [review]

File: EIPS/eip-4512.md
Comment (KKimos) [reply EIPS/eip-4512.md]: Thank you for your suggestion, I have revised it.

File: EIPS/eip-4512.md
Comment (chee-chyuan) [inline EIPS/eip-4512.md:59]: i do believe you are able to instead do this for natspec
`@param tokenId <description>`

Comment (KKimos) [reply EIPS/eip-4512.md]: Thanks again, it has been modified.

**Comment by wschwab** [comment]
There is a similar ERC also in PR right now: [EIP-4400](https://github.com/ethereum/EIPs/pull/4400). While there are certainly differences between the two standards, I was wondering if it might be worth your time to try uniting these into one standard as opposed to two very similar ones.

File: EIPS/eip-4512.md
Comment (wschwab) [inline EIPS/eip-4512.md:246]: Similar to above, I'm personally not fully understanding the intended path to leverage this EIP for NFT mortgages or rentals. The `Rationale` section is more for design decisions, so I don't know if this would be the right place to put it, but I think that at least for myself, if there would be more in the `Motivation` explaining how this EIP is intended to be leveraged, it would help me understand here.

This would also be a good section for discussing why the authors opted to create a more simplified [EIP-2612](./eip-2612.md), and any other alternatives that were considered but ultimately discarded in favour of the current EIP.

File: EIPS/eip-4512.md
Comment (wschwab) [inline EIPS/eip-4512.md:250]:
```suggestion
This protocol is fully backward compatible with ERC721. Refer to the Specification section above for more details.
```

File: EIPS/eip-4512.md
Comment (wschwab) [inline EIPS/eip-4512.md:255]: I don't see any tests - are there tests somewhere to run?

File: EIPS/eip-4512.md
Comment (wschwab) [inline EIPS/eip-4512.md:4]:
```suggestion
description: ERC721 NFTs with a separate user role
```

**Review by wschwab — COMMENTED** [review]
Hi! I mentioned in the PR thread that this EIP seems very similar imho to [EIP-4400](https://github.com/ethereum/EIPs/pull/4400), and I would recommend collaborating. In the meantime, I've left some comments that I hope are helpful.

**Comment by KKimos** [comment]
> fully

I tried to contact the author of EIP-4400 and request to merge with him

**Comment by lightclient** [comment]
> I tried to contact the author of EIP-4400 and request to merge with him

Any progress on this?

**Pull request closed without merging**
